### PR TITLE
chore(deps): upgrade jackson

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

This upgrades Jackson to the latest version (2.13.1). Previously autotests were still stuck on 1.9.x so it also required upgrading some of the code to use the new API.

Also, to address the assembly deduplication issues, I consolidated the handling of the module-info files to the root `build.sbt` and tidied up the sub-project `build.sbt` files.

Lastly, I noticed that we were running the tests (in CI) for a whole package of tests however we were still declaring the individual classes in the TestNG YAML file, so I tidied that up to.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by edalex-ian** [comment]
Starting this off as 'draft' as I worked on it a couple of weeks back and all was fine, and then have rebased it back onto develop and am getting a couple of tests failures locally (out by one). So I'm wondering if some of the changes have bumped things - especially the change around PagedSearchResults bug.

File: autotest/Tests/build.sbt [hunk 1]
```diff
@@ -20,29 +20,28 @@ libraryDependencies ++= Seq(
...
-  "org.http4s"                %% "http4s-blaze-client"      % http4sVersion,
-  "org.http4s"                %% "http4s-circe"             % http4sVersion,
-  "org.typelevel"             %% "cats-free"                % catsVersion,
-  "com.unboundid"             % "unboundid-ldapsdk"         % "6.0.3"
+  "javax.jws"                  % "javax.jws-api"             % "1.1",
+  "org.apache.commons"         % "commons-lang3"             % "3.12.0",
+  "org.seleniumhq.selenium"    % "selenium-java"             % "3.141.59",
+  "org.easytesting"            % "fest-util"                 % "1.2.5",
+  "org.easytesting"            % "fest-swing"                % "1.2.1",
+  "com.fasterxml.jackson.core" % "jackson-databind"          % jacksonVersion,
```
Comment (edalex-ian) [inline autotest/Tests/build.sbt:28]: Now using `jacksonVersion` from `project/CommonSettings.scala`.

File: build.sbt [hunk 2]
```diff
@@ -98,6 +98,15 @@ checkJavaCodeStyle := {
...
 }
 
+// We currently build for Java 8, so let's drop module info files
+ThisBuild / assemblyMergeStrategy := {
+  case PathList("module-info.class")         => MergeStrategy.discard
+  case x if x.endsWith("/module-info.class") => MergeStrategy.discard
+  case x =>
+    val oldStrategy = (ThisBuild / assemblyMergeStrategy).value
+    oldStrategy(x)
+}
```
Comment (edalex-ian) [inline build.sbt:108]: Centralisation of the deduplication handling for `module-info`.

**Comment by edalex-ian** [comment]
Will re-run CodeBuild as it only failed on a single `io.github.openequella.rest.Search2ApiTest`:

```
FAILED: collectionTest
--
java.lang.AssertionError: expected [2] but found [3]
```

That's different to what I saw locally. :thinking:

**Comment by edalex-ian** [comment]
Same single error again. Odd. Will look at it closer.

File: autotest/OldTests/src/test/java/io/github/openequella/rest/FavouriteApiTest.java [hunk 3]
```diff
@@ -35,7 +37,9 @@ public void testAddFavouriteItem() throws IOException {
     assertEquals(HttpStatus.SC_CREATED, makeClientRequest(method));
 
     JsonNode response = mapper.readTree(method.getResponseBody());
-    String[] keywords = mapper.readValue(response.get("keywords"), String[].class);
+    ObjectReader stringArrayReader = mapper.readerFor(new TypeReference<String[]>() {});
```
Comment (PenghaiZhang) [inline autotest/OldTests/src/test/java/io/github/openequella/rest/FavouriteApiTest.java:40]: Is the previous approach deprecated ?

File: autotest/Tests/build.sbt (hunk 1 above)
Comment (PenghaiZhang) [reply autotest/Tests/build.sbt:28]: But why not using val `jacksonDataBind` defined in CommonSettings.scala?

File: autotest/OldTests/src/test/java/io/github/openequella/rest/FavouriteApiTest.java (hunk 3 above)
Comment (edalex-ian) [reply autotest/OldTests/src/test/java/io/github/openequella/rest/FavouriteApiTest.java:40]: More like.... removed. :wink:

File: autotest/Tests/build.sbt (hunk 1 above)
Comment (edalex-ian) [reply autotest/Tests/build.sbt:28]: True that, that came after - as you know. :wink: 

I'll tweak.

**Comment by edalex-ian** [comment]
Interesting, it could be the order of tests as you mentioned @PenghaiZhang .

Locally I just did a green run of the full `io.github.openequella` and then attempted to run the `com.tle.webtests.test.webservices.rest` package which failed in a couple of places. Following that I ran `io.github.openequella` and had a couple of fails too.

Maybe I need to revert the changes to testng-codebuild.yaml file. :disappointed:

**Comment by edalex-ian** [comment]
Alright, with that reverted and the changes from develop (to improve the GHA build) I'm aiming for a green CodeBuild and green Old UI from GHA and then I'll take this out of draft.

**Comment by edalex-ian** [comment]
Alright, ready for review now @PenghaiZhang

**Review by PenghaiZhang — APPROVED** [review]
Thanks Ian! I think it's good to go now!

**Pull request merged**